diff --git a/simdpp/detail/preprocessor/config/config.hpp b/simdpp/detail/preprocessor/config/config.hpp
index 0bfc91f..2c2bd11 100644
--- a/simdpp/detail/preprocessor/config/config.hpp
+++ b/simdpp/detail/preprocessor/config/config.hpp
@@ -73,7 +73,7 @@
 # define SIMDPP_PP_VARIADICS_MSVC 0
 # if !defined SIMDPP_PP_VARIADICS
 #    /* variadic support explicitly disabled for all untested compilers */
-#    if defined __GCCXML__ || defined __CUDACC__ || defined __PATHSCALE__ || defined __DMC__ || defined __CODEGEARC__ || defined __BORLANDC__ || defined __MWERKS__ || ( defined __SUNPRO_CC && __SUNPRO_CC < 0x5120 ) || defined __HP_aCC && !defined __EDG__ || defined __MRC__ || defined __SC__ || defined __IBMCPP__ || defined __PGI
+#if defined __GCCXML__ || defined __CUDACC__ || defined __PATHSCALE__ || defined __DMC__ || ((defined __CODEGEARC__ || defined __BORLANDC__) && (!defined(__clang__) || __clang_major__ < 15)) || defined __MWERKS__ || (defined __SUNPRO_CC && __SUNPRO_CC < 0x5120) || defined __HP_aCC && !defined __EDG__ || defined __MRC__ || defined __SC__ || defined __IBMCPP__ || defined __PGI
 #        define SIMDPP_PP_VARIADICS 0
 #    elif defined(_MSC_VER) && defined(__clang__)
 #        define SIMDPP_PP_VARIADICS 1
